src/DenoHTTPWorker.ts

```diff
@@ -0,0 +1,377 @@
+   * environment in Deno. If specified, this script will be used instead of the
+   * default bootstrap script. Only advanced users should set this.
+   */
+  denoBootstrapScriptPath: string;
+
+  /**
+   * Flags that are passed to the Deno process. These flags follow the `deno
+   * run` command and can be used to enabled/disabled various permissions and
+   * features. These flags will be modified to ensure that the deno process can
+   * run the http server we'll be connecting to. Toggle the
+   * printCommandAndArguments if you need to understand what the final flag
+   * values are.
+   *
+   * Review Deno's available flags here:
+   * https://docs.deno.com/runtime/manual/getting_started/command_line_interface
+   */
+  runFlags: string[];
+
+  /**
+   * Print stdout and stderr to the console with a "[deno]" prefix. This is
+   * useful for debugging.
+   */
+  printOutput: boolean;
+
+  /**
+   * Print out the command and arguments that are executed.
+   */
+  printCommandAndArguments: boolean;
+
+  /**
+   * Options used to spawn the Deno child process
+   */
+  spawnOptions: SpawnOptions;
+}
+
+export const newDenoHTTPWorker = async (
+  script: string | URL,
+  options?: Partial<DenoWorkerOptions>
+): Promise<DenoHTTPWorker> => {
+  const _options: DenoWorkerOptions = Object.assign(
+    {
+      denoExecutable: "deno",
+      denoBootstrapScriptPath: DEFAULT_DENO_BOOTSTRAP_SCRIPT_PATH,
+      runFlags: [],
+      printCommandAndArguments: false,
+      spawnOptions: {},
+      printOutput: false,
+    },
+    options || {}
+  );
+
+  let networkingIsOk = false;
+  _options.runFlags = _options.runFlags.map((flag) => {
+    if (flag === "--allow-net" || flag === "--allow-all") {
+      networkingIsOk = true;
+    }
+    if (flag === "--deny-net") {
+      throw new Error(
+        "Using --deny-net without specifying specific addresses is not supported"
+      );
+    }
+    if (flag.startsWith("--deny-net") && flag.includes(LISTENING_HOSTPORT)) {
+      throw new Error(
+        `Using --deny-net with the address ${LISTENING_HOSTPORT} is not supported`
+      );
+    }
+    if (flag.startsWith("--allow-net=")) {
+      networkingIsOk = true;
+      return (flag += "," + LISTENING_HOSTPORT);
+    }
+    return flag;
+  });
+  if (!networkingIsOk) {
+    _options.runFlags.push("--allow-net=" + LISTENING_HOSTPORT);
+  }
+
+  let scriptArgs: string[];
+
+  if (typeof script === "string") {
+    scriptArgs = ["script", script];
+  } else {
+    scriptArgs = ["import", script.href];
+  }
+
+  let command = "deno";
+  if (typeof _options.denoExecutable === "string") {
+    command = _options.denoExecutable;
+  }
+
+  if (Array.isArray(_options.denoExecutable)) {
+    if (_options.denoExecutable.length === 0)
+      throw new Error("denoExecutable must not be empty");
+
+    command = _options.denoExecutable[0]!;
+    _options.runFlags = [
+      ..._options.denoExecutable.slice(1),
+      ..._options.runFlags,
+    ];
+  }
```

Comment:
Design-wise, not sure what the intention here is - you can stick flags into the `denoExecutable` option as well as into `runFlags`?

Reply:
yes, this is maybe a little premature, but it's to allow something like `docker run denoland/deno deno` to be used as deno instead of a single executable. 

I will prob remove, since adding a `string[]` option in the future would be easy